handlers/sandbox.go

```diff
@@ -43,22 +46,83 @@ type googleClaims struct {
 	jwt.StandardClaims
 }
 
+type googleOauthToken struct {
+	Access_token string
+	Id_token     string
+}
+
+type githubAccessTokenResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	Scope       string `json:"scope"`
+}
+
 var sandboxUsersCollection *mongo.Collection = db.GetCollection("sandbox_users")
 
 func (sbh SandboxHandler) Login(c *gin.Context) {
 	var body models.SandboxLoginSchema
+	var firstName string
+	var lastName string
+	var email string
 	ok := utils.Validate(c, &body, false, nil)
 	if !ok {
 		return
 	}
-	google_token := body.Google_token
-	claims, err := validateGoogleJWT(google_token)
-	if err != nil {
-		logger.Error("Login(Sandbox) error: " + err.Error())
+	token := body.Token
+	loginType := body.LoginType
+	if loginType == "google" {
+		gOuth, err := getGoogleAuthToken(token)
+		if err != nil {
+			logger.Error("Login(Sandbox) error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		claims, err := GetGoogleUser(*gOuth)
+		if err != nil {
+			logger.Error("Login(Sandbox) error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		firstName = claims.FirstName
+		lastName = claims.LastName
+		email = claims.Email
+	} else if loginType == "github" {
+		gitAccessToken, err := getGithubAccessToken(token)
+		if err != nil {
+			logger.Error("Login(Sandbox) error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		claims, err := getGithubData(gitAccessToken)
+		if err != nil {
+			logger.Error("Login(Sandbox) error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		email, _ = claims["email"].(string)
+		if email == "" {
+			temp, _ := claims["repos_url"].(string)
+			temp2 := strings.Split(temp, "https://api.github.com/users/")
+			temp3 := strings.Split(temp2[1], "/")
+			email = temp3[0]
+		}
+		fullName := strings.Split(claims["name"].(string), " ")
+		firstName = fullName[0]
+		lastName = fullName[1]
+
+	} else {
+		logger.Error("Wrong login type")
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+
+	if email == "" {
```

Comment:
I think that it is not possible to get in here without an email or without the code will throw an error. So I think it is not needed